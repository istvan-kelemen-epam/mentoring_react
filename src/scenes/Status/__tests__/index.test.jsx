import { Status } from '../index.jsx';
import renderer from 'react-test-renderer';

describe('scenes/Status', () => {
	let status;

	beforeEach(() => {
		status = Object.create(Status.prototype);
	});

	describe('.render', () => {
		it('should create empty result', () => {
			status.props = {};
			const rendered = status.render();
			const snapshot = renderer.create(rendered);
			expect(snapshot).toMatchSnapshot();
		});

		it('should create empty result', () => {
			status.props = {data: [{}, {}]};
			const rendered = status.render();
			const snapshot = renderer.create(rendered);
			expect(snapshot).toMatchSnapshot();
		});
	});
});