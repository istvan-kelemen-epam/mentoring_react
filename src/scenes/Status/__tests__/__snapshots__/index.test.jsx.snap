// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scenes/Status .render should create empty result 1`] = `
<section
  className="status"
>
  <span
    className="status__found"
  >
    
  </span>
  <span
    className="status__sort-by"
  >
    <span
      className="status__sort-by-caption"
    >
      Sort by
    </span>
    <span
      className="status__sort-by-value"
    >
      release date
    </span>
  </span>
  <span
    className="status__rating"
  >
    rating
  </span>
</section>
`;

exports[`scenes/Status .render should create empty result 2`] = `
<section
  className="status"
>
  <span
    className="status__found"
  >
    2 movies found
  </span>
  <span
    className="status__sort-by"
  >
    <span
      className="status__sort-by-caption"
    >
      Sort by
    </span>
    <span
      className="status__sort-by-value"
    >
      release date
    </span>
  </span>
  <span
    className="status__rating"
  >
    rating
  </span>
</section>
`;
